File: src/main/java/com/wxl/dyttcrawler/properties/SchedulerProperties.java
package com.wxl.dyttcrawler.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * Create by wuxingle on 2020/5/20
 * 任务队列管理
 */
@Data
@ConfigurationProperties(prefix = "crawler.scheduler")
public class SchedulerProperties {

    private Type type = Type.LOCAL;

    public enum Type {
        LOCAL,
        REDIS
    }

}

File: src/main/kotlin/com/wxl/dyttcrawler/properties/SchedulerProperties.kt
package com.wxl.dyttcrawler.properties

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * Create by wuxingle on 2021/10/11
 * 任务队列管理
 */
@ConfigurationProperties(prefix = "crawler.scheduler")
data class SchedulerProperties(
    var type: Type = Type.LOCAL
) {

    enum class Type {
        LOCAL, REDIS
    }

}

